lib/ReactViews/ExplorerWindow/Tabs/MyDataTab/AddData.jsx

```diff
@@ -103,19 +113,18 @@ const AddData = createReactClass({
       }
     }
     addUserCatalogMember(this.props.terria, promise).then(addedItem => {
-      if (addedItem && !(addedItem instanceof TerriaError)) {
-        this.onFileAddFinished(addedItem);
-      }
       this.setState({
         isLoading: false
       });
       this.props.resetTab();
+      if (addedItem && !(addedItem instanceof TerriaError)) {
+        this.onFileAddFinished([addedItem]);
+      }
     });
   },
 
-  onFileAddFinished(fileToSelect) {
-    this.props.viewState.myDataIsUploadView = false;
-    this.props.viewState.viewCatalogMember(fileToSelect);
+  onFileAddFinished(files) {
+    this.props.onFileAddFinished(files);
```

Comment:
I think it would be cleaner to remove this function entirely and just use `this.props.onFileAddFinished` in the places that this was used.